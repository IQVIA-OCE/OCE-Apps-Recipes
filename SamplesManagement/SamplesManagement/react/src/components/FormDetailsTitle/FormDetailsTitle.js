import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useTheme, Text } from '@oce-apps/apollo-react-native';
import color from 'color';

const FormDetailsTitle = ({ title, children }) => {
  const theme = useTheme();

  const containerBgColor = color(theme.colors.surface)
    .darken(theme.dark ? 0.64 : 0.05)
    .hex()

  return (
    <View style={[styles.container, { backgroundColor: containerBgColor }]}>
      <Text style={styles.title}>{title}</Text>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 4,
    paddingHorizontal: 10,
  },
  title: {
    fontSize: 16,
    lineHeight: 30,
  },
});

export default FormDetailsTitle;
