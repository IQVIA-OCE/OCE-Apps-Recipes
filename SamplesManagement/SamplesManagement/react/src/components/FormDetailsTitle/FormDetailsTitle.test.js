import FormDetailsTitle from "./FormDetailsTitle";
import { Text } from "react-native";
import React from "react";
import { render } from '@testing-library/react-native';

describe('FormDetailsTitle', () => {
  it('Should render FormDetailsTitle ', () => {
    const { getByText } = render(
      <FormDetailsTitle title="title">
        <Text>children</Text>
      </FormDetailsTitle>
    );

    expect(getByText(/title/)).toBeTruthy();
    expect(getByText(/children/)).toBeTruthy();
  });
});