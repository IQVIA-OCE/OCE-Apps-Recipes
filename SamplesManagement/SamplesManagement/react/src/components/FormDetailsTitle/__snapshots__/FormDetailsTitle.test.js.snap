// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDetailsTitle Should render FormDetailsTitle  1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f3f2f2",
      "borderRadius": 4,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4D4D4D",
        "fontSize": 16,
        "lineHeight": 30,
      }
    }
  >
    title
  </Text>
  <Text>
    Text
  </Text>
</View>
`;
